src/driver/xen.rs

```diff
@@ -164,6 +184,86 @@ impl Introspectable for Xen {
         }))
     }
 
+    fn listen(&mut self, timeout: u32) -> Result<Option<Event>, Box<dyn Error>> {
+        let fd = self.xev.xenevtchn_fd()?;
```

Comment:
you don't need to recreate the fd and the poll structure everytime.
create it once, store it, and use it here.